pandas/io/parsers/readers.py

```diff
@@ -596,9 +597,9 @@ def read_csv(
     sep: str | None | lib.NoDefault = ...,
     delimiter: str | None | lib.NoDefault = ...,
     header: int | Sequence[int] | None | Literal["infer"] = ...,
-    names=...,
-    index_col=...,
-    usecols=...,
+    names: Sequence[Hashable] | None | lib.NoDefault = ...,
+    index_col: IndexLabel | Literal[False] | None = ...,
+    usecols: Sequence[Hashable] | Callable[[Hashable], bool] | None = ...,
```

Comment:
`` Sequence[Hashable]`` allows something like ``["a", 1]`` right?

This is not strict enough here, because this actually raises.

```
ValueError: 'usecols' must either be list-like of all strings, all unicode, all integers or a callable.
```

Not sure if this is feasible to type accordingly, but wanted to mention it at least

Reply:
That is, unfortunately, the case (but also a bit inconsistent, a user could pass `names=["a", 1, 2]` and also `usecols=["a", 1]`). Would be great If we wouldn't allow any Hashable as column names.

Will probably leave usecols untyped.

Reply:
I removed the annotations for `usecols` to keep this PR simple/uncontroversial.